### PR TITLE
feat(user-modify-entry) Modify an entry by a user

### PR DESCRIPTION
**Pull request by obulaworld**
**What does this PR do?**

Setup an endpoint for a user to modify a  database stored entry

**Description of Task to be completed?**

- Setup entry point
- Run tests on the entry point

**How should this be manually tested?**
Clone the repo, cd into it and run npm test.

**Any background context you want to provide?**
No

**What are the relevant pivotal tracker stories?**
#159288506

### CODE REVIEW

File: backend-controller/entryController.js [hunk 1]
```diff
@@ -12,36 +10,93 @@ class DiaryController {
...
-  static editEntry(req, res) {
-    const entryToModify = allEntries.find(e => e.id === parseInt(req.params.id, 10));
-    if (!entryToModify || entryToModify === undefined) {
-      res.status(404).json({
-        error: 'The entry you wish to modify must have been removed or have not been created',
+    if (error) {
+      res.status(400).json({ error: error.details[0].message });
+      return;
+    }
+    const check = auth.verifyUserToken(req);
```
Comment (Dannytebj) [inline backend-controller/entryController.js:19]: @obulaworld  Based on our review yesterday, I thought we agreed that you would take all validation to the middleware, why are you still doing this here?

File: backend-controller/entryController.js [hunk 2]
```diff
@@ -12,36 +10,93 @@ class DiaryController {
...
     if (error) {
       res.status(400).json({ error: error.details[0].message });
       return;
     }
-    entryToModify.title = req.body.title;
-    entryToModify.category = req.body.category;
-    entryToModify.sub_category = req.body.sub_category;
-    entryToModify.content = req.body.content;
-    res.status(200).json({ success: 'success', entry: entryToModify });
+    const check = auth.verifyUserToken(req);
```
Comment (Dannytebj) [inline backend-controller/entryController.js:57]: @obulaworld  Please take all validation to middleware, apply every other place

**Review by Dannytebj — COMMENTED** [review]
@obulaworld Please implement the feedback I gave on refactoring validations to middleware

File: backend-controller/entryController.js (hunk 1 above)
Comment (obulaworld) [reply backend-controller/entryController.js:19]: okay....u know we agreed i will merge the current branch then create a refactoring branch

File: backend-controller/entryController.js (hunk 2 above)
Comment (obulaworld) [reply backend-controller/entryController.js:57]: okay..thanks

**Pull request merged**